### DISCUSSION TITLE
Issue with ClassDB instantiation of custom classes in Godot 4.2.2

### DISCUSSION THREAD

**Discussion by lazy-pigeon** [godotengine/godot-proposals#9626] [Engine Core]
I've encountered an issue while using Godot version 4.2.2 that has been troubling me: 
Why is it that ClassDB cannot instantiate custom classes? I want to instantiate classes by using their StringName, but it seems currently unattainable.
 I'm curious if there are alternative methods to achieve this goal or if there are plans to add support for this functionality in future releases?

![Snipaste_2024-04-29_14-44-02](https://github.com/godotengine/godot-proposals/assets/33474350/7ce3090e-b78c-4e23-bbcb-9fe2ee49cbc8)
![Snipaste_2024-04-29_14-44-30](https://github.com/godotengine/godot-proposals/assets/33474350/6d05279f-15c9-4b46-a688-d90a8d1242c9)

**Comment by theraot** [comment]
Classes created from scripts are not in the ClassDB.

As, I presume you know, if you know the name of the class while writing the code, you can use it to create instances.

However, if you only have the name in runtime, what you need is to get a reference to the script from that name. You can do so by searching for it in [`get_global_class_list`](https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-method-get-global-class-list), match the `"class"` key, and read the `"path"`, which you can then `load` to get a script object. If you know the type of script you can cast it (e.g cast to `GDScript`) and call `new` on it.

**Reply by lazy-pigeon** [reply]
Thank you so much! Your help was invaluable, and your solution worked perfectly!
